plugins/org.komodo.core/src/org/komodo/repository/LocalRepository.java

```diff
@@ -308,18 +308,27 @@ public void respond( final Object results ) {
                                                                                                   callback, getSession(),
                                                                                                   getName()));
 
+                Exception error = null;
                 boolean noTimeout = false;
 
                 try {
                     noTimeout = latch.await(3, TimeUnit.MINUTES);
                 } catch (final Exception e) {
-                    callback.equals(e);
+                    error = e;
                 }
 
                 if (noTimeout) {
-                    callback.respond(null);
+                    if (getCallback() != null) {
+                        getCallback().respond( null );
+                    }
                 } else {
-                    callback.errorOccurred(new KException(Messages.getString(Messages.LocalRepository.Commit_Timeout, getName())));
+                    if (error == null) {
+                        callback.errorOccurred( new KException( Messages.getString( Messages.LocalRepository.Commit_Timeout,
```

Comment:
I don't think `callback` can ever be `null` here as it is constructed above in line 306. Now `getCallback()` can be `null`. Naming is a bit confusing as that method does not return the `callback` instance but returns the listener passed in when the `UnitOfWork` was constructed. Am I looking at this right?